ConsistencyManager/ConsistencyManager.swift

```diff
@@ -696,6 +737,28 @@ public class ConsistencyManager {
         }
     }
 
+    // MARK: Threading
+
+    /**
+     A helper function which wraps our queue.
+     You can call the closure provided by the block to check if the operation has been cancelled.
+     */
+    private func dispatchTask(task: (()->Bool)->Void) {
+        // In order to get cancelled for an NSBlockOperation, you need to do a weak dance
+        // If this is strong, then the block will hold a reference to itself and cause a retain cycle
+        weak var weakOperation: NSBlockOperation?
+        let operation = NSBlockOperation() {
```

Comment:
my personal preference in this scenario is to define the operation without a block and then add a block to it.

```
let operation = NSBlockOperation()
operation.addExecutionBlock { [weak operation] in
}
```